pilot/pkg/model/push_context.go

```diff
@@ -827,6 +830,19 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *config.C
 		return out
 	}
 
+	if features.EnableDestinationRuleInheritance {
```

Comment:
I am a bit confused. I thought we were merging rules? This looks more like its falling back to global defaults if there is no destination rule, but the moment a DR is created the globals will be dropped?

Reply:
>the moment a DR is created the globals will be dropped?
The previous logic for retrieving the DestinationRule for a service did not need to be changed because the user created rules were merged with any global rules during init in `SetDestinationRules`.  This block of code handles the scenarios where rule look-up for a particular service has yielded no results, so first try to return the namespace rule (which was already merged with appropriate global mesh rule during init) or return the mesh rule if no namespace level rule is found

Reply:
I see, read it wrong. thanks